### DISCUSSION TITLE
Is termux dead?

### DISCUSSION THREAD

**Discussion by ranile** [termux/termux-app#3618] [Q&A] [answered]
The last release is from 2 years ago. There has been activity in the GitHub but nothing has been released. Is the project dead and not receiving any updates?

**Comment by TomJo2000** [comment, marked as answer]
0.119 will be released when it is good and ready.

Termux 0.118 is a - close to - feature complete Terminal emulator.
Since the app and packages are updated independently of one another
a new release of the App has not been in a while.

As such we have the opportunity to take our time
and focus on making continued development as sustainable as we can.
Mainly by tackling accrued technical debt, which does take time.


Several major blockers for sustainable development are being currently cleared.
e.g.
- Updating the NDK to the latest release (26).
  - termux/termux-packages#17863
- Updating Clang and LLVM toolchain to the newly released LLVM 17.
  - termux/termux-packages#17751
- and addressing the `targetSdk >= 29` code execution restrictions to be able to migrate Termux and its packages to a newer SDK version
  - #2155
  - termux/termux-exec#24

There's also various other improvements due to be merged into 0.119 like:
- Full Italic font support
  - #711
- Sixel/iTerm2 image in terminal support
  - #142
- Background image support
  - #3079
- And enhanced support for various escape sequences 
  - #2600 
  - #2656

**Reply by ranile** [reply]
That's great to hear. I came across the project today. Good to hear it's still alive and well

I suggest having a pinned issue or something that outlines the roadmap so future people can know the status (maybe even contribute!)

**Reply by kyounger** [reply]
Came here to say the same thing. As someone installing apps directly from GitHub releases, it took me finding this closed discussion thread before I was comfortable moving forward given the age of the last release. Super pumped to use this, though!

**Reply by TomJo2000** [reply]
> any updates?

This isn't a development thread.
Please don't [necro-bump](https://en.wiktionary.org/wiki/necro-bump) resolved discussions without an actual reason.
I will however oblige the request for comment.

---

Java devs with the requisite domain knowledge and time/motivation to get familiar with the Termux codebase and are willing/able to take on the burden of developing/maintaining it are an exceptionally rare find.
Progress on the app side is slow since we barely have anybody working on the main app.
And that's okay, Termux 0.118 is a nearly feature complete terminal emulator, getting some improvements like those mentioned above merged into a stable release is gonna be nice.
But nothing here is a showstopper and package/tooling updates are continuing at an accelerating pace.

0.119 will get here when it gets here, it'll probably have some nice improvements,
it'll definitely deal with some common stumbling stones like outdated default repos,
but there is no reason to rush things.
In the meantime we'll keep adding and updating packages and refining Termux specific tooling such as [Termux:X11](https://github.com/termux/termux-x11?tab=readme-ov-file#termuxx11).

**Reply by sylirre** [reply]
Perhaps I will just lock this thread.

Why are there no updates? This is not correct question in the context of OSS project development. Just because no one is obligated to deliver these updates, all work is done voluntarily.

The correct question is "What I did to help Termux development?". That question contradicts all consumer principles. Obviously, one would rather want a new fancy Termux:X11 rice instead of puzzling over new feature implementation. Imbalance between consumerism and contribution always was a problem for open source, but in case with Termux is brought to the new level. A lot of our users literally seeing the Linux shell first time in their life and not able to contribute due to low skill level...

Truth as it is.

**Reply by sylirre** [reply]
If 1% of community would contribute to Termux development on a frequent basis, we would see new releases every week.

Otherwise if community would be passive regarding to development question and put all the burden onto few persons who maintain Termux now, then everyone should be ready that gap between releases may be quite long.
